lib/matplotlib/axis.py

```diff
@@ -1229,11 +1229,13 @@ def _set_lim(self, v0, v1, *, emit=True, auto):
             self.axes.callbacks.process(f"{name}lim_changed", self.axes)
             # Call all of the other axes that are shared with this one
             for other in self._get_shared_axes():
-                if other is not self.axes:
-                    other._axis_map[name]._set_lim(
-                        v0, v1, emit=False, auto=auto)
-                    if other.figure != self.figure:
-                        other.figure.canvas.draw_idle()
+                if other is self.axes:
+                    continue
+                other._axis_map[name]._set_lim(v0, v1, emit=False, auto=auto)
+                if emit:
+                    other.callbacks.process(f"{name}lim_changed", other)
```

Comment:
True, but I think guarding the shared-axes syncing with `if emit:` is actually wrong (#26085), whereas the extra `if emit:` here is correct, so perhaps let's keep it there so that we don't forget to put it in when removing the outer check?

Reply:
I was wondering a bit about that behavior, but didn't take the time to actually test it...

Could it be as simple as dedenting the for loop (and keeping this if)?

That would:
- call the code to actually update the shared axes, regardless of `emit`
- make `emit` truly only tied to the callback behavior, which is what I would expect

Reply:
Dedenting the for loop *will* lead to infinite recursion, so that will need some kind of marker to block recursion (but it can't be done by overloading the semantics of emit as I proposed initially...).